src/elements/OcStarToggle.vue

```diff
@@ -0,0 +1,65 @@
+<template>
+  <oc-button
+    variation="raw"
+    role="switch"
+    :aria-checked="checked.toString()"
+    :aria-label="label"
+    @click="togglePressed"
+  >
+    <oc-star :shining="checked" />
+  </oc-button>
+</template>
+
+<script>
+import OcButton from "./OcButton"
+
+/**
+ *
+ * ## Accessibility
+ *
+ * ### Communicating the buttons toggle role to screen readers
+ * With `role="switch"` screen readers are informed that the button is a toggle, which will have two states: on and off. Toggle buttons like this should be used instead of checkboxes (which make only really sense in form contexts – [reasoning and more background](https://inclusive-components.design/toggle-button/#thisdoesntquitefeelright)).
+ *
+ * ### Communicating the toggle's state to screen readers
+ *
+ * Buttons with `switch` role can convey their state with the `aria-checked` attribute. When using this component, authors can set the initial state of OcStarToggle with the `checked` prop.
+ *
+ * */
+
+export default {
+  name: "oc-star-toggle",
+  components: { OcButton },
+  status: "review",
+  release: "1.0.0",
+  methods: {
+    togglePressed: function() {
```

Comment:
Add `$_ocStarToggle` prefix pls to keep the naming convention in ODS. (see e. g. https://github.com/owncloud/owncloud-design-system/blob/5cf4b0bdc69a9f32bddc2d90a082e08f9e961938/src/patterns/OcAutocomplete.vue#L214)